clang/lib/Basic/Targets/X86.h

```diff
@@ -835,16 +835,22 @@ class LLVM_LIBRARY_VISIBILITY UEFIX86_64TargetInfo
 public:
   UEFIX86_64TargetInfo(const llvm::Triple &Triple, const TargetOptions &Opts)
       : UEFITargetInfo<X86_64TargetInfo>(Triple, Opts) {
+    LongWidth = LongAlign = 32;
+    DoubleAlign = LongLongAlign = 64;
+    IntMaxType = SignedLongLong;
+    Int64Type = SignedLongLong;
+    SizeType = UnsignedLongLong;
+    PtrDiffType = SignedLongLong;
+    IntPtrType = SignedLongLong;
+    LongDoubleWidth = LongDoubleAlign = 64;
+    LongDoubleFormat = &llvm::APFloat::IEEEdouble();
```

Comment:
I understand that this was copied from the MSVC target but it's worth pointing out that neither [UEFI specification](https://uefi.org/specs/UEFI/2.11/02_Overview.html#data-types) nor the [TianoCore reference implementation](https://github.com/tianocore/edk2/blob/master/MdePkg/Include/X64/ProcessorBind.h) mention anything about floating point types so it's not clear whether this is correct or not.

Reply:
I've asked someone who does work quite often with EFI stuff and they don't know either. It sounds like it's up to the developer to handle it. It's possible it's not even allowed.

Reply:
I'll update the patch removing the floating point support related LongDoubleFormat. I'll add inline comments as well to reflect that we assume defaults for integer types, alignment, C++ abi etc based on what Windows target sets as default. Our goal is to support non-default cases for these down the road as well and I'll try to express that in the comment.